src/Types.hs

```diff
@@ -88,38 +139,63 @@ data Category =
   deriving (Eq, Ord, Bounded, Enum, Show, Read, Generic)
 
 -- | "Verbesserungsvorschlag"
+--
+-- 'Comments' are hierarchical.  The application logic is responsible for putting some limit on the
+-- recursion depth under which all children become siblings.  (FIXME: or should we represent this in
+-- the types?)
 data Comment = Comment
-    { _commentMeta    :: MetaInfo
-    , _commentArticle :: Article
-    , _commentVotes   :: Set Vote
+    { _commentMeta    :: MetaInfo Comment
+    , _commentText    :: Document
+    , _commentVotes   :: Set Comment
+    , _commentReplies :: Set CommentVote
     }
   deriving (Eq, Ord, Show, Read, Generic)
 
--- | "Stimme"
-data Vote = Vote
-    { _voteMeta  :: MetaInfo
-    , _voteValue :: Maybe Bool
+-- | "Stimme" for "Idee"
+data CommentVote = CommentVote
+    { _commentVoteMeta  :: MetaInfo CommentVote
+    , _commentVoteValue :: Bool
     }
   deriving (Eq, Ord, Show, Read, Generic)
 
 data User = User
-    { _userMeta           :: MetaInfo
-    , _userName           :: ST
-    , _userPassword       :: EncryptedPass
-    , _userEmail          :: Maybe Email
+    { _userMeta     :: MetaInfo User
+    , _userLogin    :: ST
+    , _userFullName :: (ST, ST)
```

Comment:
I think we should either have a `FullName` datatype or two inlined fields for first and last name.